tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -313,30 +313,44 @@ class UnsortedSegmentBaseOp : public OpKernel {
     auto output_flat = output->flat_outer_dims<T>();
 
     auto data_ptr = data.template flat<T>().data();
-    ReductionFunction(context, context->template eigen_device<Device>(), output_rows,
+      this->operator()(context, context->template eigen_device<Device>(), output_rows,
         segment_ids.shape(), segment_flat, data.NumElements(), data_ptr,
         output_flat);
   }
-private:
-  Functype ReductionFunction;
 };
 
 template <typename Device, class T, class Index>
 class UnsortedSegmentSumOp
-  : public UnsortedSegmentBaseOp<Device, T, Index,
-                                   functor::UnsortedSegmentSumFunctor<Device, T, Index>> {
+  : public UnsortedSegmentBaseOp<Device, T, Index> {
 public:
   explicit UnsortedSegmentSumOp(OpKernelConstruction* context)
-    : UnsortedSegmentBaseOp<Device, T, Index, functor::UnsortedSegmentSumFunctor<Device, T, Index>>(context, functor::UnsortedSegmentSumFunctor<Device, T, Index>()) {}
+    : UnsortedSegmentBaseOp<Device, T, Index>(context) {}
```

Comment:
Like described before, when passing sum and max functors as arguments, it will use the basefunctor, instead of the specialized functors. For the same reason multiple inheritance is used.
I got it to work with pointers now in the new commit.